packages/client/src/pages/SignIn/SignIn.tsx

```diff
@@ -0,0 +1,105 @@
+import { useForm } from 'react-hook-form';
+import { Navigate } from 'react-router-dom';
+
+import { FC, useState } from 'react';
+import { AuthApi } from '../../api/Auth/Auth';
+import { SignInDto } from '../../api/Auth/types';
+import AppLink from '../../components/AppLink/index';
+import FormButton from '../../components/FormButton';
+import Input from '../../components/Input';
+import { Navbar } from '../../components/Navbar';
+import { FormValues } from './types';
+import { validateLogin, validatePassword } from './validationUtils';
+
+const authApi = new AuthApi();
+
+const submit = (dto: SignInDto) => {
+  return authApi.SignIn(dto);
+};
+
+export const SignIn: FC = () => {
+  const [hasLoggedIn, setHasLoggedIn] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: {
+      errors: { root, login: loginError, password: passwordError },
+      isSubmitting,
+    },
+    reset,
+  } = useForm<FormValues>({
+    mode: 'onChange',
+  });
+
+  const onSubmit = async (data: FormValues) => {
+    try {
+      await submit(data);
```

Comment:
Почему бы тут просто не сделать вызов authApi напрямую:
```suggestion
      await authApi.SignIn(data);
```
В чём смысл добавить ещё один слой?

Reply:
Второй момент что у вас нигде не указано что SignIn это асинхронная функция, а тут делаете await

Reply:
Имею в виду, что не хватает именно ключевое слово "async"

Reply:
дак я внутри SignIn не использую await, поэтому не объявлял её как асинхронную. Она просто промис возвращает